ssl/ssl_sess.c

```diff
@@ -319,8 +320,12 @@ int ssl_generate_session_id(SSL *s, SSL_SESSION *ss)
     }
 
     /* Choose which callback will set the session ID */
-    CRYPTO_THREAD_read_lock(s->lock);
-    CRYPTO_THREAD_read_lock(s->session_ctx->lock);
+    if (!CRYPTO_THREAD_read_lock(s->lock))
+        return 0;
+    if (!CRYPTO_THREAD_read_lock(s->session_ctx->lock)) {
+        CRYPTO_THREAD_unlock(s->lock);
+        return 0;
```

Comment:
It should probably call SSLfatal()

Reply:
added SSLfatal call, with `SSL_R_SESSION_ID_CONTEXT_UNINITIALIZED`  not perfect but i didn't want to add an error code.